### DISCUSSION TITLE
npm run serve get type error but vscode with volar is fine

### DISCUSSION THREAD

**Discussion by youlinaa** [pnpm/pnpm#4269] [Q&A]
**pnpm version v6.14.5**

I used package `vue-docgen-api`，and `vue-docgen-api` has these dependencies：

![image](https://user-images.githubusercontent.com/32128165/150461742-4a0f6938-e77d-43e1-af69-c142c5bc1a68.png)

if open my project on vscode and try `ctrl + click` package's name `vue-docgen-api`，it will resolved to `node_modules/.pnpm/vue-docgen-api@4.35.0/node_modules/vue-docgen-api`, type is right.

![image](https://user-images.githubusercontent.com/32128165/150461930-d045f549-89a9-486b-9d48-4f8a51426876.png)

![image](https://user-images.githubusercontent.com/32128165/150486895-7e11cbb7-4ce4-40a1-b569-34483bdc7d9c.png)

but when run command that `npm run serve`, will got err:

![image](https://user-images.githubusercontent.com/32128165/150462263-f7015692-af07-46c2-a5c8-c2cb9c7e54fc.png)

it seem like resolve to package is different ? 

[here](https://github.com/xinnianyoulin/test-pkg-resolve) have a repository about this question.

**Comment by zkochan** [comment]
Try with `node-linker=hoisted`

After setting this in `.npmrc`, remove node_modules and run `pnpm install` again.

**Reply by youlinaa** [reply]
Thanks reply. but it's not work. 

![image](https://user-images.githubusercontent.com/32128165/150626373-c7c64667-b08c-4266-aeb3-cc28e2869407.png)

I found that when set `"preserveSymlinks": true` in tsconfig.json,  then retry `ctrl + click` package's name `vue-docgen-api` will got same result like `npm run serve`.

Is the problem related to webpack's resolve.symlinks?

**Reply by zkochan** [reply]
if you use the "isolated" linker (which is the default), then symlinks should not be preserved

**Reply by youlinaa** [reply]
Yeah, I just wanted to show that when `npm run serve`, it's like set "preserveSymlinks": true, I actually set "preserveSymlinks": false.

**Reply by youlinaa** [reply]
Can you reproduce the problem with the repository I provided?
